### PR TITLE
Fix stateful traits availability check not considering Squeak

### PR DESCRIPTION
**Pull request by tom95**

### CODE REVIEW

File: MonticelloTonel-Core.package/TonelParser.class/instance/hasStatefulTraits.st [hunk 1]
```diff
@@ -1,5 +1,5 @@
 private testing
 hasStatefulTraits
-	"Pharo has stateful traits starting on version 7"
+	"Pharo has stateful traits starting on version 7, Squeak does not have them at all"
 	
-	^ SystemVersion current major >= 7
\ No newline at end of file
+	^ SystemVersion current baseName ~= 'Squeak' and: [SystemVersion current major >= 7]
```
Comment (j4yk) [inline MonticelloTonel-Core.package/TonelParser.class/instance/hasStatefulTraits.st]: #baseName comes from Etoys-Squeakland and is not available in Pharo. Is there a way that relies only on the basics?

In Pharo there is #type, but not in Squeak. In Squeak there is #isSqueak, but not in Pharo... hmm. Looks like the safest way is to duplicate some of the logic and just check the #version string.

**Review by j4yk — CHANGES_REQUESTED** [review]

**Comment by tom95** [comment]
Oh man great catch, thank you @j4yk! I adapted it to use `version beginsWith:` and verified that it works in Pharo. The check in general feels pretty shady, I wonder if there might be a `Smalltalk at: ...` we could use instead.

**Review by j4yk — APPROVED** [review]

**Comment by j4yk** [comment]
./configure style would be to compile a test program ;-)

For now this looks good enough.

**Comment by j4yk** [comment]
Rebased (squashed the whitespace fix commit) and integrated in the squeak branch.

**Pull request closed without merging**